### ISSUE TITLE
Save button bugged in Agent Scheduler's image viewer

### ISSUE THREAD

**Issue by afbagwell** [ArtVentureX/sd-webui-agent-scheduler#140]
This applies to SDNext/vladmandic. I don't know if A1111 is affected.

The save button inside the viewer instance provided by the Agent Scheduler doesn't work.

I verified that none of the images saved this way actual get copied to the destination. No "Copying image: ..." output in the console. 

If you use the image viewer from the "From Text" tab in SDNext the save button works as expected.

**Comment by artventuredev** [comment]
Can you share a screen capture about the button. I don't remember the extension provides any Save button.

**Comment by afbagwell** [comment]
From the Task History, click on a preview image to go to the image viewer

<img width="1432" alt="Screenshot 2023-10-01 013105" src="https://github.com/ArtVentureX/sd-webui-agent-scheduler/assets/1495201/acdb6e65-0db6-41eb-8f04-ae420887ef93">

Once you are in the viewer, the fourth button from the left at the top should save the image to your assigned save directory path. But clicking it doesn't do anything.

<img width="1145" alt="Screenshot 2023-10-01 013142" src="https://github.com/ArtVentureX/sd-webui-agent-scheduler/assets/1495201/3a844d22-76ac-4489-9f7c-e324919f758e">

This same image viewer is using on the text2img and img2img tabs to look at your generated output. The save button in those cases works properly.

I originally posted this issue to vladmandic, but his response was that is was an issue with the extension.

https://github.com/vladmandic/automatic/issues/2265

**Comment by artventuredev** [comment]
On A1111 there's no Save button in the image viewer, so I think this only available in SDNext. I'll need to take a look on how SDNext is handling this.

![image](https://github.com/ArtVentureX/sd-webui-agent-scheduler/assets/133728487/ad4a3ab0-ddc9-426b-98e0-3c2593775ed0)

**Comment by artventuredev** [comment]
By the way, what behavior are you expecting when you click the `Save` button in the task history review? The images should have already been saved to the disk.

**Comment by afbagwell** [comment]
True, however the practical use of the save button is to re-save an image you want to keep by placing another copy in a specific folder defined by the setting:

"Image Naming & Paths" -> "Directory for saving images using the Save button" 

I use this all the time if I generate say 50 images, flip though them hitting the save button on the few I think are worthwhile to keep. Then I can delete everything in the original generation folder. Otherwise I would to use some other app like MS Photos or IrfanView to cycle though the images and delete everything I didn't want while keeping the few I did, which is way more tedious.

It's also a bit confusing that the same image viewer window is used for txt2img, img2img and Agent Scheduler but the save button doesn't work in the latter. If you use it in the other two, here's an example of the console output:

<img width="848" alt="Screenshot 2023-10-01 093227" src="https://github.com/ArtVentureX/sd-webui-agent-scheduler/assets/1495201/f6e72865-7955-4170-88b2-30d6a0d2f0cd">

**Comment by artventuredev** [comment]
I understand your point. The original implementation functions only when you're on one of the three main tabs (Text, Image, Process), where additional Save buttons are present in the generation result area. Eg:
![image](https://github.com/ArtVentureX/sd-webui-agent-scheduler/assets/133728487/4b6d55e3-534a-433f-bff9-ab27aa25ac58)

To implement this, I'll need to add an extra Save button to the agent scheduler tab. Given that this isn't a critical issue, I plan to address it next weekend, as my current workload from my day job is quite substantial.
